packages/eslint-plugin/docs/rules/strict-boolean-expressions.mdx

```diff
@@ -55,6 +56,11 @@ let obj = {};
 while (obj) {
   obj = getObj();
 }
+
+// assertion functions without an `is` are boolean contexts.
+declare function myAssert(value: unknown): asserts value;
+let maybeString = Math.random() > 0.5 ? '' : undefined;
+myAssert(maybeString);
```

Comment:
[Docs] Nit: everywhere else generally refers to `assert()`, and that's a pretty commonly used name. I think it'd be nice to use it here for consistency.

```suggestion
declare function assert(value: unknown): asserts value;
let maybeString = Math.random() > 0.5 ? '' : undefined;
assert(maybeString);
```

Reply:
hmm, this intentional. I was hoping to emphasize that this applies to any user-defined assertion functions, not just special-casing a built-in (or some yucky function name heuristic).

Will change though if you think sticking to `assert()` is better

Reply:
I think the fact that it explicitly shows `function assert` (even after a `declare`) is enough to indicate it's any user-declared one. I think anybody who's understanding docs about assertion functions should understand this.

Reply:
sounds good 👍 Changed